ui/src/main/webapp/app/common/csv-parser-service.js

```diff
@@ -183,5 +183,58 @@ angular.module('app').factory('csv', function() {
         return processed;   // once parsed, we can return the parts.
     }
 
+    /**
+     * Generates CSV based on a an array of objects, and a header array.
+     */
+    service.generate = function(rows, headers) {
+
+        var csvString = "";
+
+        if (!headers) { 
+            error.handle('Unable to parse CSV with no headers');
+        }
+
+        for (var i in headers) {
+            if (headers[i].indexOf(',') !== -1) {
+                csvString += '"' + headers[i] + '",';
+            } else {
+                csvString += headers[i] + ',';
+            }
+        }
+
+        csvString = csvString.substr(0, csvString.length - 1);
+        csvString += '\r\n';
+        
+
+        for (var i in rows) {
+            var row = rows[i];
+
+
+            for (var j in headers) {
+                var field = row[headers[j]];
+                
+                if (field === null || field === undefined) {
+                    csvString += ',';
+                    continue;
+                }
+
+                if (typeof field === 'string' && field.indexOf(',') !== -1) {
```

Comment:
I think you should start by replacing any " with "". Then wrap the field in " " if there are any commas. Otherwise you may run into an issue with a field like:
```
hello",world
```

Replacing " with "" gives:

```
hello"",world
```

Then wrap it in " " as there is a comma:
```
"hello"",world"
```

Perhaps add a test for this too.